be1-go/hub/standard_hub/hub_state/ids_by_channel.go

```diff
@@ -0,0 +1,59 @@
+package hub_state
+
+import (
+	"golang.org/x/exp/slices"
+	"maps"
+	"sync"
+)
+
+// IdsByChannel provides a thread-safe structure that stores a channel id with its corresponding message ids
+type IdsByChannel struct {
+	sync.RWMutex
+	table map[string][]string
+}
```

Comment:
this could then become 
```suggestion
type IdsByChannel ThreadSafeMap[string,[]string]
```